### ISSUE TITLE
20 Epochs on COCO - (Larger Transformer)

### ISSUE THREAD

**Issue by afiaka87** [lucidrains/DALLE-pytorch#336]
### Discussed in https://github.com/lucidrains/DALLE-pytorch/discussions/335

<div type='discussions-op-text'>

<sup>Originally posted by **afiaka87** July 11, 2021</sup>
[Full W&B training session](https://wandb.ai/dalle-pytorch-replicate/COCO512_16_16D_16H_80TSL)

![media_images_image_14100_c563c7f9470a4a3dd2c2](https://user-images.githubusercontent.com/3994972/125195299-b1814c00-e21a-11eb-8642-4e010dd8d113.png)

![media_images_image_14500_d5fdc93c3d9bba882b25](https://user-images.githubusercontent.com/3994972/125195303-b3e3a600-e21a-11eb-9b69-a3da7b075875.png)

![coco_trained](https://user-images.githubusercontent.com/3994972/125195424-248ac280-e21b-11eb-8231-cd9cede6d549.png)

Details
Transformer:
 - Visual Dim - 512
 - Max Text Length/Language Dim - 80
 - Depth - 16
 - Heads - 16
 - Head Dim - 64
 - Attention: (axial_row, axial_row, axial_col, axial_row,axial_row, axial_row, axial_col, axial_row,axial_row, axial_row, axial_col, full,axial_row, axial_row, axial_col, full) # note the two layers of dense attention
 - lr_decay = True
 - Reversible = False
 
Hardware:
- 1x RTX 2070 Super (8 GiB VRAM)
- 1x AMD Ryzen 3900
- 32 GiB RAM

Checkpoints:
You can find a checkpoint for each epoch trained here:
https://wandb.ai/dalle-pytorch-replicate/COCO512_16_16D_16H_80TSL/artifacts/model/trained-dalle/07c445559fd9183e302e</div>
